lib/templateFactory.js

```diff
@@ -321,6 +345,63 @@ class TemplateFactory extends BaseFactory {
         return super.get(config);
     }
 
+    /**
+     * Get metrics for the pipelines that use the given template
+     * @method getPipelineUsage
+     * @param  {String}     fullTemplateName    Name of the template and the version or tag (e.g. chef/publish@1.2.3)
+     * @return {Promise}                        List of metrics for the pipelines
+     */
+    async getPipelineUsage(fullTemplateName) {
+        const template = await this.getTemplate(fullTemplateName);
+        const { id } = template;
+        const metricPipelinesConfig = {
+            params: { templateId: id },
+            readOnly: true,
+            aggregationField: 'pipelineId'
+        };
+
+        // eslint-disable-next-line global-require
+        const JobFactory = require('./jobFactory');
+        const jobFactory = JobFactory.getInstance();
+
+        const pipelineIds = await jobFactory
+            .list(metricPipelinesConfig)
+            .then(pipelines => pipelines.map(p => p.pipelineId));
+
+        // eslint-disable-next-line global-require
+        const PipelineFactory = require('./pipelineFactory');
+        const pipelineFactory = PipelineFactory.getInstance();
+
+        const pipelineConfig = {
+            params: { id: pipelineIds },
+            readOnly: true
+        };
+
+        // eslint-disable-next-line global-require
+        const EventFactory = require('./eventFactory');
+        const eventFactory = EventFactory.getInstance();
+
+        const pipelineData = await pipelineFactory.list(pipelineConfig);
+
+        return Promise.all(
+            pipelineData.map(async p => {
+                const eventConfig = {
+                    params: { id: p.lastEventId }
+                };
+                const possibleEvent = await eventFactory.list(eventConfig);
+                const event = possibleEvent.length > 0 ? possibleEvent[0].createTime : null;
+
+                return {
+                    id: p.id,
+                    name: p.name,
+                    scmRepo: p.scmRepo,
+                    lastRun: event,
+                    admins: p.admins
+                };
+            })
```

Comment:
Per discussion, instead of making individual call to EventFactory for each pipeline, let iterate over the pipelines to collect all the eventIds and make just one call to DB